reproject/interpolation/core_celestial.py

```diff
@@ -23,7 +25,8 @@ def _reproject_celestial(array, wcs_in, wcs_out, shape_out, order=1):
     """
 
     # Make sure image is floating point
-    array = np.asarray(array, dtype=float)
+    if not np.issubdtype(array.dtype, np.float):
+        array = np.asarray(array, dtype=float)
```

Comment:
Good point. 

But `float` is `np.float64` while `f4` (even ignoring endianess) is `np.float32` (or even `np.float16`, I never got the hang of those 😄). Not sure if you need/want double precision for reproject.